sdk/servicebus/service-bus/src/receivers/shared.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ContextWithSettlement } from "../models";
+import { ServiceBusMessage } from "../serviceBusMessage";
+import { throwErrorIfClientOrConnectionClosed } from "../util/errors";
+import { ClientEntityContext } from "../clientEntityContext";
+import { RuleDescription, CorrelationFilter } from "../core/managementClient";
+
+/**
+ * @internal
+ * @ignore
+ */
+export function getSubscriptionRules(context: ClientEntityContext): Promise<RuleDescription[]> {
+  if (context.entityPath.includes("/Subscriptions/")) {
```

Comment:
I get that we don't have any other way to distinguish if an entity is a queue/subscription 
(which was one of the motivations for me to add an entity_type property.)

Maybe I'm nit-picking at this point, but let's consider this scenario with a queue name - `level1/Subscriptions/level2`.
Such a queuename is totally possible and I could even send messages to it.
![image](https://user-images.githubusercontent.com/10452642/76577609-30c60a80-6483-11ea-8ed5-757f4fb2da7e.png)

This `.includes("/Subscriptions/")` validation is not enough to determine if it is a queue/subscription in this case.

Reply:
Ha, you're absolutely right. If you do it in the portal it warns you about it and provides this page:

https://docs.microsoft.com/en-us/archive/blogs/servicebus/azure-service-bus-azure-resource-manager-and-this-character

Also, I didn't write this code so it's always been there, lurking with this potential bug. :)

Reply:
No, it wasn't there from the beginning. I wrote it in a previous iteration but was hoping to differentiate with `entityType: "queue" | "subscription"`.

Reply:
Hahaha, okay. You trapped me then!

Reply:
My only concern with the `entityType: "queue" | "subscription"` is that the user can pass the formatted path of "topic+subscription"(instead of separate topicname and subscription name) to get the receiver and the sdk would classify the `formatted("topic+subscription")` as a queue.

Reply:
Filed #7815

Reply:
Thanks.